ca_candidates/people.py

```diff
@@ -28,17 +34,128 @@ def scrape(self):
+                ),
+                "image": (
+                    lambda p: p.image,
+                    lambda p, value: setattr(p, "image", value),
+                ),
+            }
+
         # parties being scraped
         parties = (
             "liberal",
             "ndp",
             "green",
+            "conservative",
         )
 
         for party in parties:
             party_method = getattr(self, f"scrape_{party}")
             for method in party_method():
                 p = method
+                if crowdsourcing.get(key):
+                    o = crowdsourcing[key]
+
+                    links = {}
+                    for link in p.links:
+                        domain = ".".join(urlsplit(link["url"]).netloc.split(".")[-2:])
+                        if domain in ("facebook.com", "fb.com"):
+                            links["facebook"] = link["url"]
+                        elif domain == "instagram.com":
+                            links["instagram"] = link["url"]
+                        elif domain == "linkedin.com":
+                            links["linkedin"] = link["url"]
+                        elif domain == "twitter.com":
+                            links["twitter"] = link["url"]
+                        elif domain == "youtube.com":
+                            links["youtube"] = link["url"]
+
+                    for prop, (getter, setter) in steps.items():
+                        if o[prop]:
+                            if prop == "email" and ".gc.ca" in o[prop]:
+                                self.info(f"{key}: skipping email = {o[prop]}")
+                            else:
+                                scraped = getter(p)
+                                if not scraped:
+                                    setter(p, o[prop])
+                                    self.debug(f"{key}: adding {prop} = {o[prop]}")
+                                elif scraped.lower() != o[prop].lower() and prop != "image":
+                                    self.warning(f"{key}: expected {prop} to be {scraped}, not {o[prop]}")
+
+                    for prop in ["facebook", "instagram", "linkedin", "twitter", "youtube"]:
+                        if o[prop]:
+                            scraped = links.get(prop)
+                            entered = re.sub(
+                                r"/timeline/\Z|\?(f?ref|lang|notif_t)=.+|\?_rdr\Z",
+                                "",
+                                o[prop].replace("@", "").replace("http://twitter.com/", "https://twitter.com/"),
+                            )  # Facebook, Twitter
+                            if not scraped:
+                                p.add_link(entered)
+                                self.debug(f"{key}: adding {prop} = {entered}")
+                            elif scraped.lower() != entered.lower():
+                                self.warning(f"{key}: expected {prop} to be {scraped}, not {entered}")
```

Comment:
I suggest starting an entirely new branch, and just committing the scrape_conservative method and this section here (and the one above).